src/middlewared/middlewared/api/base/decorator.py

```diff
@@ -2,12 +2,20 @@
 import functools
 
 from .handler.accept import accept_params
+from ..base.model import BaseModel
 from middlewared.schema.processor import calculate_args_index
 
 __all__ = ["api_method"]
 
 
-def api_method(accepts, returns, audit=None, audit_callback=False, audit_extended=None, roles=None):
+def api_method(
+    accepts: type[BaseModel],
+    returns: type[BaseModel],
+    audit: str | None = None,
+    audit_callback=False,
```

Comment:
Let's add type hints here as well as args before/after have them

Reply:
`audit` has a boolean default which implies its type, and `audit_callback` can't have a type hint, because it's a callable with non-specific arguments